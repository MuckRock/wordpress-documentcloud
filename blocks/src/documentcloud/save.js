/**
 * WordPress dependencies.
 */
import { useBlockProps } from '@wordpress/block-editor';

/**
 * Save component for the DocumentCloud block.
 *
 * @param {Object} props            Block props.
 * @param {Object} props.attributes Block attributes.
 * @return {JSX.Element} Block save element.
 */
export default function Save( { attributes } ) {
	const { embeddedHtml } = attributes;

	// Get block props
	const blockProps = useBlockProps.save();

	// If there's no embedded HTML yet, return empty container
	if ( ! embeddedHtml ) {
		return <div { ...blockProps }></div>;
	}

	/**
	 * The embedded HTML is trusted code from the oEmbed endpoint so we can directly set it dangerously in the innerHTML.
	 * This allows the oEmbed code to directly embed in page from server and prevent nesting it inside an iframe.
	 */
	return (
		<div { ...blockProps }>
			<div
				className="documentcloud-embed"
				dangerouslySetInnerHTML={ { __html: embeddedHtml } }
			></div>
		</div>
	);
}
