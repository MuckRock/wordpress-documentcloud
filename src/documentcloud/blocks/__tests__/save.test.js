import React from 'react';
import save from '../src/documentcloud/save';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

// Mock the useBlockProps.save function from @wordpress/block-editor
jest.mock( '@wordpress/block-editor', () => ( {
	useBlockProps: {
		save: jest.fn( () => ( {
			className: 'wp-block-documentcloud',
		} ) ),
	},
} ) );

describe( 'DocumentCloud Block Save Component', () => {
	it( 'renders an empty container when no embeddedHtml is provided', () => {
		const attributes = {
			embeddedHtml: '',
		};

		const { container } = render( save( { attributes } ) );

		// Ensure the container is rendered with the correct className
		expect( container.firstChild ).toHaveClass( 'wp-block-documentcloud' );
		// Ensure the container is empty
		expect( container.firstChild ).toBeEmptyDOMElement();
	} );

	it( 'renders the embedded HTML when embeddedHtml is provided', () => {
		const attributes = {
			embeddedHtml:
				'<iframe src="https://embed.documentcloud.org/documents/23060035-notes"></iframe>',
		};

		const { container } = render( save( { attributes } ) );

		// Ensure the container is rendered with the correct className
		expect( container.firstChild ).toHaveClass( 'wp-block-documentcloud' );
		// Ensure the embedded HTML is rendered inside the container
		expect(
			container.querySelector( '.documentcloud-embed' )
		).toBeInTheDocument();
		expect(
			container.querySelector( '.documentcloud-embed' ).innerHTML
		).toBe(
			'<iframe src="https://embed.documentcloud.org/documents/23060035-notes"></iframe>'
		);
	} );

	// add snapshot test
	it( 'matches the snapshot', () => {
		const attributes = {
			embeddedHtml:
				'<iframe src="https://embed.documentcloud.org/documents/23060035-notes"></iframe>',
		};

		const { container } = render( save( { attributes } ) );

		expect( container ).toMatchSnapshot();
	} );
} );
