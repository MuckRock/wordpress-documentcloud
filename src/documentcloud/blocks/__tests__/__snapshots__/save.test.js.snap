// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentCloud Block Save Component matches the snapshot 1`] = `
<div>
  <div
    class="wp-block-documentcloud"
  >
    <div
      class="documentcloud-embed"
    >
      <iframe
        src="https://embed.documentcloud.org/documents/23060035-notes"
      />
    </div>
  </div>
</div>
`;
